sound/soc/sof/ipc4-topology.c

```diff
@@ -31,6 +31,8 @@ static const struct sof_topology_token ipc4_sched_tokens[] = {
 		offsetof(struct sof_ipc4_pipeline, use_chain_dma)},
 	{SOF_TKN_SCHED_CORE, SND_SOC_TPLG_TUPLE_TYPE_WORD, get_token_u32,
 		offsetof(struct sof_ipc4_pipeline, core_id)},
+	{SOF_TKN_SCHED_PIPELINE_KCPS, SND_SOC_TPLG_TUPLE_TYPE_WORD, get_token_u32,
```

Comment:
@lyakh, @ujfalusi  and @jsarha I had a chat with @plbossart and can we modify this attribute to be per module i.e. needs to be in the base class of each module. This would then allow us to mix scheduling types in a pipeline and scale across cores. 

Thinking further, can we also extend this to include format and be an array i.e.
```c
{SOF_TKN_SCHED_MODULE_KCPS_S16LE, SND_SOC_TPLG_TUPLE_TYPE_ARRAY, get_token_u32_array,
```
Where the array index would be the channel count + 1. i.e. when defined in topology
```c
1000, 2000, 0, 4000, 0, 0, 0, 0
```
This would then specify 1000 kcps for mono S16_LE, 2000kcps for stereo S16_LE, 3 channel S16_LE not known (can use platform max default), and so on.

Reply:
IIRC the CPC/CPS are kind of worst-case figures. They are not meant to be adjusted depending on channel count or input/output formats. We're probably looking at one broadbrush estimate for each module and correction factor to account for the known unknowns (cache, IPC, solar eclipses and neutrinos).